File: leetcode/0078-Subsets/solution.go
func subsets(nums []int) [][]int {
    powerSet := make([][]int, 0)
    powerSet = append(powerSet, []int{})

    for _, v := range(nums) {
        lenSet := len(powerSet)
        for i := 0; i < lenSet; i++ {
            copySet := make([]int, len(powerSet[i]))
            copy(copySet, powerSet[i])
            copySet = append(copySet, v)
            powerSet = append(powerSet, copySet)
        }
    }

    return powerSet
}

File: leetcode/0078-Subsets/solution.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        powerSet = []
        powerSet.append([])

        for v in nums:
            lenSet = len(powerSet)
            for i in range(lenSet):
                powerSet.append(powerSet[i].copy())
                powerSet[-1].append(v)

        return powerSet